### PR TITLE
session: Support protocol changes related to sidechain magic

### PR DESCRIPTION
**Pull request by cthulhu-rider**

### CODE REVIEW

File: pkg/client/opts.go [hunk 1]
```diff
@@ -88,11 +90,21 @@ func WithBearer(token *token.BearerToken) CallOption {
 	}
 }
 
+// WithSidechainMagic returns option to specify network magic of NeoFS sidechain.
+//
+// Zero by default.
+func WithSidechainMagic(magic uint64) CallOption {
```
Comment (alexvanin) [inline pkg/client/opts.go:96]: Wouldn't it be quite a pain to provide `clients.WithSidechainMagic` in every request we send in neofs-node? Maybe we will benefit from `WithDefaultSideChainMagic` client option as well?

**Review by alexvanin — COMMENTED** [review]
Should neofs-node check side chain magic before signature check?

If yes, then we can ignore version negotiation and the client can set magic regardless of the remote node version.

If no, then we have to ignore side chain magic field if remote node has API <v2.10. Otherwise signature check will fail.

File: pkg/client/opts.go (hunk 1 above)
Comment (cthulhu-rider) [reply pkg/client/opts.go:96]: I made it a `CallOption` after thought "oh, magic is a request field, so it is a per-call value". But in current implementation we expect that storage node can't switch magic number within opened connection, so we can change it to `Option`.

**Comment by cthulhu-rider** [comment]
There is a challenge in version compatibility control. After discussion we've decided to postpone sidechain magic feature after the error system's intro.

**Comment by alexvanin** [comment]
This PR not broken, though, unlike others. We can keep it, in this case.

**Comment by alexvanin** [comment]
Nevermind, it is broken too :face_exhaling:

**Pull request closed without merging**